src/avram/bulk_upsert.cr

```diff
@@ -0,0 +1,93 @@
+class Avram::BulkUpsert(T)
+  def initialize(@records : Array(T),
+                 @conflicts : Array(Symbol),
+                 @permitted_fields : Array(Symbol))
```

Comment:
Probably don't need to set this as the instance variable since you're overwriting it 3 lines later